wandb/sdk/wandb_init.py

```diff
@@ -913,6 +913,7 @@ def _attach(
 
 
 def init(
+    job_name: Optional[str] = None,
```

Comment:
Why do you need it as a new arg in init? We're trying not to be adding these unless absolutely necessary. We've been putting such stuff to Settings, so `wandb.init(settings=wandb.Settings(job_name=...))` would have the same effect. One caveat is documenting the Settings object, but we're getting there :)

Reply:
Adding that kwarg is the main purpose of this PR. To make it easier for users to set a name for their job when it is created. We can just put it in settings but I don't think most users know that the settings object even exists. I think a wandb.init kwarg is far more discoverable and clear.

Reply:
@dmitryduev can you help me understand why you prefer not to introduce new `wandb.init` kwargs unless absolutely necessary?

Reply:
We are removing the argument. Job name will be configurable by settings and WANDB_JOB_NAME until we revamp the API